homework/index.js

```diff
@@ -29,14 +29,130 @@
+    const optionArr = data.map(repository => repository.name);
+    const optionsArr = [];
+    for (let i = 0; i < optionArr.length; i++) {
+      const option = {
+        name: optionArr[i],
+        index: i,
+      };
+      optionsArr.push(option);
+    }
+    optionsArr
+      .sort((a, b) => a.name.localeCompare(b.name, { caseFirst: 'lower' }))
+      .map(element =>
+        createAndAppend('option', selectMenu, {
+          value: element.index,
+          text: element.name,
+          id: element.name,
+        }),
+      );
+    return header;
+  }
+
+  function ulMaker(data) {
+    const rightDiv = document.querySelector('.rightDiv');
+    createAndAppend('p', rightDiv, { class: 'contributor-header', text: 'Contributions' });
+    const ul = createAndAppend('ul', rightDiv, { class: 'contributor-list' });
+    data.forEach(contributor => {
+      const li = createAndAppend('li', ul, {
+        class: 'contributor-item',
+        tabindex: '0',
+        'aria-label': contributor.login,
+      });
+      createAndAppend('img', li, {
+        src: contributor.avatar_url,
+        class: 'contributor-avatar',
+        height: '48px',
+      });
+      const contributorData = createAndAppend('div', li, { class: 'contributor-data' });
+      createAndAppend('div', contributorData, { text: contributor.login });
+      createAndAppend('div', contributorData, {
+        class: 'contributor-badge',
+        text: contributor.contributions,
+      });
+      li.onclick = () => {
+        window.open(contributor.html_url, '_blank');
+      };
+    });
+  }
+
+  function constructForFirst(data, index) {
+    document.querySelector('.leftDiv').innerHTML = '';
+    tableMaker(data, index);
+    fetchJSON(data[index].contributors_url, (err, fetchedData) => {
+      const root = document.getElementById('root');
+      if (err) {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      } else {
+        ulMaker(fetchedData);
+      }
+    });
+  }
+
+  function htmlConstructor(data) {
+    headerMaker(data);
+    const root = document.getElementById('root');
+    const container = createAndAppend('div', root, { id: 'container' });
+    createAndAppend('div', container, { class: 'leftDiv whiteframe' });
+    createAndAppend('div', container, { class: 'rightDiv whiteframe' });
+    const selectMenu = document.querySelector('select');
+    constructForFirst(data, selectMenu.options[selectMenu.selectedIndex].value);
+    selectMenu.onchange = () => {
+      document.querySelector('.leftDiv').innerHTML = '';
+      document.querySelector('.rightDiv').innerHTML = '';
+      const index = selectMenu.options[selectMenu.selectedIndex].value;
+      tableMaker(data, index);
+      fetchJSON(data[index].contributors_url, (err, fetchedData) => {
+        if (err) {
+          createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        } else {
+          ulMaker(fetchedData);
+        }
+      });
```

Comment:
This code here seems a repetition of what your function `constructForFirst` already does. I would think it should be possible to create a function that you can reuse for the first repo (e.g. `repositories[0]` and the selected repository (`repositories[index]`).

Reply:
yes. I was confused and I was stuck. than I solved the 'start' problem with a completely separate function. 
now I corrected that. and again I say, 'I understood things those I did not understand while I was writing'.
thanks for all Jim.